packages/aws-cdk-lib/aws-cloudfront/test/response-headers-policy.test.ts

```diff
@@ -180,4 +180,26 @@ describe('ResponseHeadersPolicy', () => {
       },
     });
   });
+
+  test('it respects CSP `reportOnly` flag by mapping to custom header', () => {
```

Comment:
This is only testing a single logic path. We need to test each of the paths added by this PR.

Reply:
This test hits both new paths introduced by this PR
- `reportOnly` being set
- `customHeadersBehavior` not existing